db/forward_iterator.cc

```diff
@@ -53,6 +53,12 @@ class ForwardLevelIterator : public InternalIterator {
     if (pinned_iters_mgr_ && pinned_iters_mgr_->PinningEnabled()) {
       pinned_iters_mgr_->PinIterator(file_iter_);
     } else {
+      /*
+      TODO(AR) needed for ASSERT_STATUS_CHECKED in
+      MergeOperatorPinningTest/MergeOperatorPinningTest.TailingIterator
+      */
+      status_.PermitUncheckedError();
+
```

Comment:
The problem is that status_ is never set (or checked) by this test.  A better solution is to put the PermitUncheckedError into the constructor.  That way, if the status is ever set and not checked by another test, they would see the assert.

I believe that is safe to do (and probably the right thing) without the associated comment.